### ISSUE TITLE
F-Droid Support

### ISSUE THREAD

**Issue by UltimateOmega** [stenzek/duckstation#1104]
Would you consider adding Duckstation to F-Droid ?

**Comment by TheEvilSkeleton** [comment]
https://gitlab.com/fdroid/rfp/-/issues/1588

**Comment by stenzek** [comment]
Closing since I have no intention to pay f-droid to list my free app. Honestly, that requirement seems very silly to me - charging to list your free and open-source application?

This project has already cost me a few hundreds out of pocket and I'd like to avoid spending any more, considering it doesn't make any money (and I'd prefer to keep it that way). At least with the Play Store I get some value out of the developer account - crash reports, analytics, etc.

If someone else wants to list it, I'm not opposed to it - the apks on GitHub are exactly the same binaries which are pushed to the Play Store. But it's not something I'm going to do myself. But if you're building yourself, please do not claim to be an official source, and I can't be expected to support potentially modified releases.

**Comment by TheEvilSkeleton** [comment]
> Closing since I have no intention to pay f-droid to list my free app. Honestly, that requirement seems very silly to me - charging to list your free and open-source application?

Where did you get this from? You never have to pay for F-Droid, let alone they build everything from source and publish it themselves. Also see comment above.

**Comment by stenzek** [comment]
> Where did you get this from? You never have to pay for F-Droid, let alone they build everything from source and publish it themselves. Also see comment above.

In the linked issue: 
![image](https://user-images.githubusercontent.com/11288319/105139264-cfaaa100-5b41-11eb-9af9-0a60677085ed.png)

**Comment by stenzek** [comment]
> let alone they build everything from source and publish it themselves

Also not really a fan of this :/ Trusting a third party to build the application in an environment I don't control can lead to issues due to different build systems, compiler versions, etc. Hence what I said about it being fine, but should not be an official release as I can't be expected to support something I can't control.

**Comment by lionirdeadman** [comment]
> > Where did you get this from? You never have to pay for F-Droid, let alone they build everything from source and publish it themselves. Also see comment above.
> 
> In the linked issue:
> ![image](https://user-images.githubusercontent.com/11288319/105139264-cfaaa100-5b41-11eb-9af9-0a60677085ed.png)

Donation is not necessary. After all, it couldn't be called a donation if it was mandatory. I believe it's mostly to show solidarity and encourage people to donate.

> Also not really a fan of this :/ Trusting a third party to build the application in an environment I don't control can lead to issues due to different build systems, compiler versions, etc. Hence what I said about it being fine, but should not be an official release as I can't be expected to support something I can't control.

I believe this is all documented here : https://f-droid.org/en/docs/

**Comment by stenzek** [comment]
> Donation is not necessary. After all, it couldn't be called a donation if it was mandatory. I believe it's mostly to show solidarity and encourage people to donate.

A checkbox makes it sound mandatory :/

**Comment by lionirdeadman** [comment]
Looking at other applications which were merged, that's not the case but I'll relay the feedback if someone hasn't already.

**Comment by IzzySoft** [comment]
@stenzek it is NOT mandatory. Check through the existing and already processed issues, as @lionirdeadman already did. Honestly, I myself (disclosure: I'm one of the F-Droid maintainers) intentionally ignore that box entirely.

Please don't ask me how it got there (I only noticed it when it showed up). As maintaining your app, maintaining F-Droid is quite a lot of work – plus has its costs, so it runs on donations. Consider that box the same as the huge banner showing up at Wikipedia from time to time: an annoyance :speak_no_evil:

**TL;DR:** you can safely ignore it. Its status will have no effect on whether or how your issue will be processed.

So go ahead, file your RFP. I'd even say take me by my word concerning processing – and yes, as far as the RFP is concerned you can do that; I just cannot attend the entire inclusion process as I'm not familiar with building the languages your project is using (I can handle Java and Kotlin apps mostly fine, Flutter meanwhile to a degree – but everything else I've got to leave to my team-mates).

**Comment by stenzek** [comment]
> As maintaining your app, maintaining F-Droid is quite a lot of work – plus has its costs, so it runs on donations. Consider that box the same as the huge banner showing up at Wikipedia from time to time: an annoyance 🙊

That's understandable, and I'm not saying it's wrong and you don't have expenses :) I just got the impression it was like the play store, and that you essentially had to pay to list free apps, which imo is kinda lame. But thanks for clarifying :)

As for building, it's nothing special - just your big standard gradle build script: https://github.com/stenzek/duckstation/blob/74650e91fa6a37ded59ca0677243e2e873bf6ffa/.github/workflows/rolling-release.yml#L183-218

Not really super interested in setting it up myself as I already have my hands full with multiple platforms, but if anyone else wants to do it and anything needs merging upstream, I'm fine with that.

**Comment by IzzySoft** [comment]
> I just got the impression it was like the play store, and that you essentially had to pay to list free apps,

If someone tried that at F-Droid, the :hankey: would hit the fan. While we certainly cannot live on air and love, pressing that way would be the same as replacing the "F" by… well, take your pick :rofl:

> which imo is kinda lame

That's an euphemism. I'd call it ridiculous. It already looks strange as it's now; suggestions for rephrasing (which emphasize its being optional but highly appreciated – you know what I mean) are welcome!

> As for building

I can only "copy-paste" in this case, so I better leave it to someone with the technical understanding. The person opening the request with us was "ghosted" (i.e. they deleted their account), so I don't know who could take over that part. Ideally someone familiar with your app (so one of the contributors here).

> Not really super interested in setting it up myself as I already have my hands full with multiple platforms

Ask me! :rofl:

> if anyone else wants to do it and anything needs merging upstream, I'm fine with that.

Thanks! I've updated that check-box in the RFP then. Looking forward to see it progressing then! As far as I can, I'll help with it – but that's double-limited in this case, as pointed out.

**Comment by TheEvilSkeleton** [comment]
> I can only "copy-paste" in this case, so I better leave it to someone with the technical understanding. The person opening the request with us was "ghosted" (i.e. they deleted their account), so I don't know who could take over that part. Ideally someone familiar with your app (so one of the contributors here).

Oh yeah, the user that opened the issue was "TheMainOne", which was my alt, just under a different username. I'm not a developer/contributor, though.

**Comment by IzzySoft** [comment]
Ah, I see. Well, I've linked to here from there leaving a corresponding statement. The "games" section isn't exactly my speciality, neither the languages used by the app, so it will probably be another maintainer/contributor working on that (I just jumped in to clear the misunderstanding caused by our unfortunate phrasing). Would be nice to have someone on the issue "over there" to keep the contact to "over here"; ideally a contributor here, but again that's not mandatory. Main point is we got the misunderstanding cleared up, Connor agreeing – so there's no stopper I can see to process it further. Of course someone (from our team at F-Droid) must find time to process it, and I'm drowning already…

**Comment by PookaMustard** [comment]
@IzzySoft if I want to take on the mantle of managing things between here and F-Droid, what do I need to know or do? I'm currently considering it. Also I think it's possible to offer updates for both of DuckStation's branches (preview, development) simultaneously in one store listing, correct?

**Comment by IzzySoft** [comment]
> what do I need to know or do?

Basically: as a contributor here you're (at least to a degree) familiar with the code here (and for missing parts know whom to address). So you'd need to help our packagers to get things working – initially, and if it ever should get stuck with updates. So when issues arise at F-Droid which need help from the team here to be solved, we'd call out to you (which with some luck will be the "rare exception"). Ideally, you also keep track if any updates are not picked up within reasonable time (say if you make a release here, and it didn't show up at F-Droid even 2..3 weeks later, there's most likely something stuck).

> Also I think it's possible to offer updates for both of DuckStation's branches (preview, development) simultaneously in one store listing, correct?

Yes, that's possible in multiple ways. As you indicate "a single listing" (and not a separate development listing), the most common approach is to use specific tag-name patterns, so only stable releases are picked-up automatically (and are then "suggested" to the user), while previews/dev-builds are submitted manually. Easier (and less work for both sides) is having dedicated listings with dedicated package names, so all of them are picked automatically from their corresponding tags (e.g. `release-<versionName>` is a stable release tag, `dev-<versionName>` is a dev build with a `.dev` appended to its package name, and so on).

**Comment by PookaMustard** [comment]
Excellent, so probably all we're gonna have to deal with is the occasional build error, and setting up the build environment. After that, I assume from what you're saying that updates are going to be fire-and-forget, so not much effort on our part.

I suppose we could give it a trial run.

**Comment by IzzySoft** [comment]
I suppose, yes :smile: You could either pick up that RFP, or start over with a fresh one. In the latter case, please reference the old one (and maybe let me know so I can close it). Once everything is green there, you might even try your luck with a merge request yourself (otherwise, one of our team will help you with that).

**Comment by stenzek** [comment]
Closing since the android sources will not be public for much longer, and as I understand it, f-droid must build their own packages, so I don't see how this will work. I'm really uncomfortable with having to do this, but it's the only way I can stop people from selling the stuff I give away for free.

**Comment by IzzySoft** [comment]
So another free choice destroyed by greed. Sad. But your move is understandable; the alternative would be chasing those greedy folks with lawyers (you could approach e.g. EFF for help on this). After all, they are violating your license.

(Btw: things can go worse. I found one of my projects being sold, and buyers turned to me for support as the seller didn't reply. Unfortunately, the buyer was sitting where my chances chasing them with a lawyer were quite thin – and I didn't know about EFF back then).

Should your decision be final, we should close the [corresponding RFP](https://gitlab.com/fdroid/rfp/-/issues/1588) as well. Please let me know.

**Comment by stenzek** [comment]
Sorry to hear about that, that's a really crappy situation to be in.

> the alternative would be chasing those greedy folks with lawyers

While possible in theory, good luck actually enforcing it. This project brings in net zero income  (actually costs me to run) and that's not something I can afford to pay for out of pocket. While I could just say "screw it, who cares", it's a real motivation killer for something I do as a hobby.

It's more of a caution thing. I haven't caught anyone selling the android frontend yet, but I'm sure it'll happen soon enough, if it isn't already. There's enough ad infested pcsxr ripoffs on the play store. However, another duckstation frontend is being used in a commercial product without my permission, secured $200k in funding, and when it's GPL licenced I can't do anything about it (they even brag about ripping off GPL software). These two frontends are at the greatest risk, nobody wants to sell a desktop PS1 emu :)

Fortunately it's an easy change, only have to rewrite a couple of things, the rest of the code is mine. 

> Should your decision be final, we should close the corresponding RFP as well. Please let me know.

Yes, please go ahead and close it. My sincere apologies for wasting everyone's time here.

**Comment by IzzySoft** [comment]
> when it's GPL licenced I can't do anything about it

Oh, sure you can! You can force them to deliver their app on the same terms, as that's what GPL demands. That's why I wrote maybe FSF can help out, they're familiar with this kind of stuff. And I vaguely remember several such cases where they succeeded (though I cannot pin-point them at the moment). Asking never hurts (unless you ask for hurt). And asking EFF wouldn't cause any costs. Might even bring in some of those 200k to your purse for "settling the issue outside courts". Or a support contract. Or both :smile:

> I suppose I could tag the latest open commit and you could build off that, but honestly I'm not sure if it's worth distributing old versions.

Right, that's something we usually avoid (exceptions are very rare). You could publish the last APK here, at its tag maybe.

OK, I'll close the RFP then (we can always reopen, should the chance arise). I understand your decision too well (though I chose otherwise back then – the project is still alive and kicking; mine, I mean, the thief/ripper has gone with the money made).
